docs/faq/code-analysis/how-long-does-it-take-for-my-repository-to-be-analyzed.md

```diff
@@ -1,7 +1,29 @@
 # How long does it take for my repository to be analyzed?
 
-Codacy usually takes under 5 minutes to analyze your repository, however it may take longer as this depends on the size of your repository.
+Codacy usually takes under 5 minutes to analyze your repository. It may however take longer, depending on a number of factors:
 
-Codacy relies on post-commit hooks sent by your Git provider to trigger the analysis after each push to the repository, so if your analysis is taking a lot of time to start [check that the Post-Commit Hook integration for your repository is enabled](../../repositories-configure/integrations/post-commit-hooks.md).
+-   **Whether it's the initial analysis of your repository**
 
-Besides this, you can also [check Codacy's status page](https://status.codacy.com/) to see if there is any ongoing incident that could delay the analysis.
+    The initial analysis examines all files in the repository, while each subsequent commit triggers an analysis only on files changed in that commit.
+
+-   **Whether tool configurations have been updated**
+
+    Updates to tool configurations trigger a reanalysis on all files in the repository at the next commit and may impact analysis duration.
+
+-   **The size of your repository**
+
+    To speed up the analysis, [ignore any files and directories](../../repositories-configure/ignoring-files.md) that aren't relevant to your project, such as generated code or any third-party libraries included in your repositories.
+
+-   **The time it takes your Git provider to trigger the analysis**
+
+    Codacy relies on post-commit hooks sent by your Git provider to trigger the analysis after each push to the repository, so if your analysis is taking a lot of time to start [check that the Post-Commit Hook integration for your repository is enabled](../../repositories-configure/integrations/post-commit-hooks.md).
+
+-   **The priority of your analysis request**
+
+    Open-source projects have lower priority in Codacy's analysis queues.
+
+-   **Whether Codacy or your Git provider is currently experiencing issues or outages**
+
+    Check [Codacy's status page](https://status.codacy.com/) and the status page of your Git provider ([GitHub](https://www.githubstatus.com/), [GitLab](https://status.gitlab.com/), [Bitbucket](https://bitbucket.status.atlassian.com/)) to see if there is any ongoing incident that could delay the analysis.
+
+-   **The current load on Codacy's servers**
```

Comment:
This is now the only point without a detailed explanation. I'm open to suggestions, but I'm OK with not disclosing more details here. I'm placing it last purely for aesthetic reasons.

Reply:
Would it make sense to gather this bullet with the priority of the analysis request? 🤔 
Although it's not completely related, I can find some connection between the two... Just for the sake of avoiding a bullet with no details.
Something like:
"- The priority of your analysis request and the current load on Codacy's servers
Open-source projects have lower priority in Codacy's analysis queues." 

Just for thought. If you think it's completely unrelated, it's good as is. 😉

Reply:
Good point and an elegant solution 😄